data/schemas/validate_playerResponse.json

```diff
@@ -0,0 +1,241 @@
+{
+	"definitions": {},
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"$id": "https://example.com/object1688732225.json",
+	"title": "Root",
+	"type": "object",
+	"required": [
+		"isSelf",
+		"name",
+		"steamID64",
+		"steamInfo",
+		"gameInfo",
+		"customData",
+		"tags"
+	],
+	"properties": {
+		"isSelf": {
+			"$id": "#root/isSelf",
+			"title": "Isself",
+			"type": "boolean",
+			"examples": [
+				false
+			],
+			"default": true
+		},
+		"name": {
+			"$id": "#root/name",
+			"title": "Name",
+			"type": "string",
+			"default": "None",
+			"examples": [
+				"Lilith"
+			],
+			"pattern": "^.+$"
+		},
+		"steamID64": {
+			"$id": "#root/steamID64",
+			"title": "Steamid64",
+			"type": "integer",
+			"default": 0,
+			"examples": [
+				76561198210264393
+			],
+			"pattern": "^\\d+$"
+		},
+		"steamInfo": {
+			"$id": "#root/steamInfo",
+			"title": "Steaminfo",
+			"anyOf": [
+				{
+					"type": "object",
+					"required": [
+						"accountName",
+						"pfp",
+						"pfphash",
+						"vacBans",
+						"isFriend",
+						"timecreated",
+						"loccountrycode"
+					],
+					"properties": {
+						"accountName": {
+							"$id": "#root/steamInfo/accountName",
+							"title": "Accountname",
+							"type": "string",
+							"default": "",
+							"examples": [
+								"Lilith"
+							],
+							"pattern": "^.*$"
+						},
+						"pfp": {
+							"$id": "#root/steamInfo/pfp",
+							"title": "Pfp",
+							"type": "string",
+							"default": "",
+							"examples": [
+								"https://avatars.akamai.steamstatic.com/427ef7d5f8ad7b21678f69bc8afc95786cf38fe6_full.jpg",
+								"https://avatars.steamstatic.com/427ef7d5f8ad7b21678f69bc8afc95786cf38fe6_full.jpg"
+							],
+							"pattern": "^.*$"
+						},
+						"pfphash": {
+							"$id": "#root/steamInfo/pfphash",
+							"title": "Pfphash",
+							"type": "string",
+							"default": "",
+							"examples": [
+								"f556979c2d55a633a063ff2b7b1eae1d2c2812fc"
+							],
+							"pattern": "^.*$"
+						},
+						"vacBans": {
+							"$id": "#root/steamInfo/vacBans",
+							"title": "Vacbans",
+							"type": "string",
+							"default": "",
+							"examples": [
+								""
+							],
+							"pattern": "^.*$"
+						},
+						"isFriend": {
+							"$id": "#root/steamInfo/isFriend",
+							"title": "Isfriend",
+							"type": "boolean",
+							"examples": [
+								true
+							],
+							"default": false
+						},
+						"timecreated": {
+							"$id": "#root/steamInfo/timecreated",
+							"title": "Timecreated",
+							"type": "integer",
+							"examples": [
+								1570311509
+							],
+							"default": 0
+						},
+						"loccountrycode": {
+							"$id": "#root/steamInfo/loccountrycode",
+							"title": "Loccountrycode",
+							"type": "string",
+							"default": "",
+							"examples": [
+								"AU"
+							],
+							"pattern": "^.*$"
+						}
+					}
+				},
+				{
+					"type": "null"
+				}
+			]
+		},
+		"gameInfo": {

```

Comment:
If we are letting the frontend request users who aren't in the game, `gameInfo` might be null.

Reply:
gameInfo should be omitted for the player history

Reply:
We could potentially include it for whatever match they were in last

Reply:
Could we calculate an "average ping" for the player history then?

Reply:
I was more thinking the player would already have the gameInfo attached when they leave the server and get put into the history list, so we could just include whatever it last was. But if it would be useful to have an average ping throughout their time that would be possible, just not a priority atm.

Reply:
Confirmed `gameInfo` has been changed to be optionally null.

Reply:
presumedly any calls to `mac/history/v1` will just be pulling cached data right? so gameInfo would just be whatever the last saved update was, and steamInfo would be the same. But being able to lookup any steamID when they haven't been cached and arent in your game would result in `gameInfo` being null. Perhaps there is something to be said about "if gameInfo == null, then do not return anything until steamInfo api call returns" (i.e. try to avoid ever returning a player object with both steamInfo and gameInfo being NULL)

Reply:
Yes I was also considering a request for a specific steam account would wait for a steam api response to come back so `steamInfo` could be populated.